unit-tests/live/syncer/test-throughput.cpp

```diff
@@ -0,0 +1,248 @@
+                //REQUIRE_NOTHROW(_ir_sensor.set_option(RS2_OPTION_EXPOSURE, 18000)); // set exposure value to get fps = 30
+        }
+        void start_streaming(configuration test)
+        {
+            _ir_sensor.set_option(RS2_OPTION_EXPOSURE, 1);
+            _ir_sensor.open(_ir_stream_profile); // ir streams in all configurations
+            _ir_sensor.start(_sync);
+            if (test == IR_RGB_EXPOSURE)
+            {
+                _rgb_sensor.open(_rgb_stream_profile);
+                _rgb_sensor.start(_sync);
+            }
+        }
+        void stop_streaming(configuration test)
+        {
+            _ir_sensor.stop();
+            _ir_sensor.close();
+            if (test == IR_RGB_EXPOSURE)
+            {
+                _rgb_sensor.stop();
+                _rgb_sensor.close();
+            }
+            unset_ir_exposure();
+        }
+        void process_validate_frames(configuration test)
+        {
+            auto t_start = std::chrono::system_clock::now();
+            auto t_end = std::chrono::system_clock::now();
+
+            float prev_fps = 0.0f;
+            auto process_frame = [&](const rs2::frame& f)
+            {
+                std::lock_guard<std::mutex> lock(_mutex);
+                auto stream_type = std::string(f.get_profile().stream_name());
+                // Only IR fps is relevant for this test, IR1 and IR2 have same fps so it is enough to get only one of them
+                if (!(stream_type == "Infrared 1" || stream_type == "Color"))
+                    return;
+                if (!f.supports_frame_metadata(RS2_FRAME_METADATA_ACTUAL_FPS) || !f.supports_frame_metadata(RS2_FRAME_METADATA_FRAME_TIMESTAMP))
+                    return;
+                auto frame_num = f.get_frame_number();
+                auto actual_fps = (float)f.get_frame_metadata(RS2_FRAME_METADATA_ACTUAL_FPS);
+                auto frame_arrival = f.get_frame_metadata(RS2_FRAME_METADATA_FRAME_TIMESTAMP); // usec
+                if (stream_type == "Infrared 1")
+                {
+                    _ir_frames_arrival_info.push_back({ (float)frame_arrival, actual_fps });
+                    if (actual_fps != prev_fps) // exposure new value is finally set
+                    {
+                        auto t_now = std::chrono::system_clock::now();
+                        _delta = (float)std::chrono::duration_cast<std::chrono::seconds>(t_now - t_start).count();
+                    }
+                    prev_fps = actual_fps;
+                }
+                if (stream_type == "Color")
+                    _rgb_frames_arrival_info.push_back({ (float)frame_arrival, actual_fps });
+            };
+
+            float delta = (float)std::chrono::duration_cast<std::chrono::seconds>(t_end - t_start).count();
+            while (delta < RECEIVE_FRAMES_TIME)
+            {
+                if (delta > RECEIVE_FRAMES_TIME/2)
+                    set_ir_exposure(); // set exposure to 18000 after 5 sec
```

Comment:
The exposure needs to be modified only once. This will create application latency and skew the results analysis